packages/expo/cli/start/server/metro/MetroTerminalReporter.ts

```diff
@@ -0,0 +1,223 @@
+      const time = chalk.dim(Date.now() - startTime + 'ms');
+      // iOS Bundling complete 150ms
+      return color(platform + status) + time;
+    }
+
+    const localPath = path.relative('.', progress.bundleDetails.entryFile);
+    const filledBar = Math.floor(progress.ratio * MAX_PROGRESS_BAR_CHAR_WIDTH);
+
+    const _progress = inProgress
+      ? chalk.green.bgGreen(DARK_BLOCK_CHAR.repeat(filledBar)) +
+        chalk.bgWhite.white(LIGHT_BLOCK_CHAR.repeat(MAX_PROGRESS_BAR_CHAR_WIDTH - filledBar)) +
+        chalk.bold(` ${(100 * progress.ratio).toFixed(1)}% `) +
+        chalk.dim(`(${progress.transformedFileCount}/${progress.totalFileCount})`)
+      : '';
+
+    return (
+      platform +
+      chalk.reset.dim(`${path.dirname(localPath)}/`) +
+      chalk.bold(path.basename(localPath)) +
+      ' ' +
+      _progress
+    );
+  }
+
+  _logInitializing(port: number, hasReducedPerformance: boolean): void {
+    // Don't print a giant logo...
+    this.terminal.log('Starting Metro Bundler');
+  }
+
+  shouldFilterClientLog(event: {
+    type: 'client_log';
+    level: 'trace' | 'info' | 'warn' | 'log' | 'group' | 'groupCollapsed' | 'groupEnd' | 'debug';
+    data: unknown[];
+  }): boolean {
+    return isAppRegistryStartupMessage(event.data);
+  }
+
+  /** Print the cache clear message. */
+  transformCacheReset(): void {
+    logWarning(
+      this.terminal,
+      chalk`Bundler cache is empty, rebuilding {dim (this may take a minute)}`
+    );
+  }
+
+  /** One of the first logs that will be printed */
+  dependencyGraphLoading(hasReducedPerformance: boolean): void {
+    // this.terminal.log('Dependency graph is loading...');
+    if (hasReducedPerformance) {
+      this.terminal.log(
+        chalk.red(
+          'Metro is operating with reduced performance.\n' +
+            'Please fix the problem above and restart Metro.\n\n'
+        )
+      );
+    }
+  }
+
+  _logBundlingError(error: SnippetError): void {
+    const moduleResolutionError = formatUsingNodeStandardLibraryError(this.projectRoot, error);
+    if (moduleResolutionError) {
+      return this.terminal.log(maybeAppendCodeFrame(moduleResolutionError, error.message));
+    }
+    return super._logBundlingError(error);
+  }
+}
+
+/**
+ * Formats an error where the user is attempting to import a module from the Node.js standard library.
+ * Exposed for testing.
+ *
+ * @param error
+ * @returns error message or null if not a module resolution error
+ */
+export function formatUsingNodeStandardLibraryError(projectRoot: string, error: SnippetError) {
+  if (!error.message) {
+    return null;
+  }
+  const { targetModuleName, originModulePath } = error;
+  if (!targetModuleName || !originModulePath) {
+    return null;
+  }
+  const relativePath = path.relative(projectRoot, originModulePath);
+
+  const DOCS_PAGE_URL =
+    'https://docs.expo.dev/workflow/using-libraries/#using-third-party-libraries';
+
+  if (NODE_STDLIB_MODULES.includes(targetModuleName)) {
```

Comment:
What happens if the module is imported like `require('node:http')`? Does this still match? (also very easy to just match for `^node:` in that case)